crates/bevy_ecs/src/world/identifier.rs

```diff
@@ -27,6 +30,29 @@ impl WorldId {
     }
 }
 
+// SAFETY: Has read-only access to shared World metadata
+unsafe impl ReadOnlySystemParam for WorldId {}
+
+// SAFETY: A World's ID is immutable and fetching it from the World does not borrow anything
+unsafe impl SystemParam for WorldId {
+    type State = WorldId;
```

Comment:
```suggestion
    type State = ();
```
We get passed the world in `get_param` so we might aswell just get the id from there

Reply:
And return `()` in `init_state`?